### ISSUE TITLE
[ISSUE] Issue with `databricks_job` resource

### ISSUE THREAD

**Issue by thaiphv** [databricks/terraform-provider-databricks#2582]
### Configuration
```hcl
data "aws_iam_instance_profile" "job_cluster" {
  name = "job_cluster_profile"
}

resource "databricks_job" "child" {
  name = "child"

  job_cluster {
    job_cluster_key = "job_cluster"
    new_cluster {
      ## ....
    }
    task {
      task_key = "task"
      job_cluster_key = "job_cluster"
      notebook_task {
        notebook_path = "/notebook"
      }
    }
  }
}

resource "databricks_job" "parent" {
  name = "parent"

  task {
    task_key = "parent"

    run_job_task {
      job_id = databricks_job.child.id
    }
  }
}
```

### Expected Behavior
The workflows should have been created without any errors.

### Actual Behavior
The workflows were created but some errors were also shown. After that, any attempts to rerun `terraform apply` failed with the same error message.

```
Plan: 1 to add, 0 to change, 0 to destroy. 
databricks_job.parent: Creating... 
Error: cannot read job: json: cannot unmarshal number into Go struct field RunJobTask.settings.tasks.run_job_task.job_id of type string 
  with databricks_job.parent, 
  on workflow.tf line 97, in resource "databricks_job" "parent": 
  97: resource "databricks_job" "parent" { 
```

### Steps to Reproduce
1. `terraform apply`

### Terraform and provider versions
Terraform: 1.4.6
Databricks provider: 1.23.0

### Debug Output
https://gist.github.com/thaiphv/61eee7d36a7f2cd5499bd4bbfef23b14

**Comment by alexott** [comment]
@gaborratky-db @geekswaroop there is a mismatch between the API (where `job_id` is int64) and TF implementation (where `job_id` is a string).

**Comment by majidfn** [comment]
We ran into the same issue. To avoid it with the current version, we had to wrap the injected job id with a `tostring` function. When this PR is merged and makes it the correct behavior, we need to fix it.

**Comment by ticklemedaly** [comment]
Stumbled upon this error myself today.

**Comment by geekswaroop** [comment]
The fix has been [merged](https://github.com/databricks/terraform-provider-databricks/pull/2588). It will be deployed later this week. Please comment here if you still face issues after the next release :)

**Comment by AlexandrVolok** [comment]
Thanks, the fix solved the issue we observed in 1.23.
